behaviours/smarc_bt/smarc_bt/bt/conditions.py

```diff
@@ -129,4 +127,53 @@ def update(self) -> Status:
         self.feedback_message = f"({plan.seconds_to_timeout}) to timeout"
         if plan.timeout_reached: 
             return Status.FAILURE
-        return Status.SUCCESS
\ No newline at end of file
+        return Status.SUCCESS
+    
+class C_TaskIsMoveTo(Behaviour):
+    def __init__(self, wara_ps_task_handler: WaraPSTaskHandler):
+        """
+        Returns S if the current task is a move_to task
+        """
+        self._wara_ps_task_handler = wara_ps_task_handler
+        name = f"{self.__class__.__name__}({self._wara_ps_task_handler})"
+        super().__init__(name)
+
+    def update(self) -> Status:
+
+        current_executing_tasks = self._wara_ps_task_handler.get_executing_tasks()
+
+        # if no tasks are executing, return failure
+        if len(current_executing_tasks) == 0:
+            self.feedback_message = "No tasks executing"
+            return Status.FAILURE
+        # focus only on first task. #TODO: this needs to be changed later, when we want multiple tasks to happen simultaneously
+        if current_executing_tasks[0]["task"]["name"] == "move-to":
+            self.feedback_message = f"Current task is move-to"
+            return Status.SUCCESS
+        else:
+            self.feedback_message = f"Not a move-to task"
+            return Status.FAILURE
+    
+class C_TaskIsRunning(Behaviour):
```

Comment:
Same idea as the TaskIsMoveTo. you can also take the "expected status" as a parameter and check for it.